app/src/store.js

```diff
@@ -0,0 +1,243 @@
+        records: [],  // Array of records that are currently being displayed.
+        filters: {},  // Map of currently selected filters.
+    },
+    mutations: {
+
+        updateTypes(state, types) {
+            /*
+             * Update the array of available RecordTypes.
+             *
+             * @param {array} types - An array of RecordTypes, formatted as JSON
+             *                        blobs, typically returned from the Grout
+             *                        API.
+             */
+            state.types = types;
+        },
+
+        updateActiveTypeId(state, type) {
+            /*
+             * Set the active (currently selected) RecordType.
+             *
+             * @param {string} type - The UUID for the RecordType object that should be
+             *                        set to `active`.
+             */
+            state.activeTypeId = type;
+        },
+
+        updateRecords(state, records) {
+            /*
+             * Set the array of currently-displayed Records.
+             *
+             * @param {array} records - An array of Records, formatted as JSON
+             *                          blobs, typically returned from the Grout
+             *                          API.
+             */
+            state.records = records;
+        },
+
+        updateFilter(state, payload) {
+            /*
+             * Update the active filters based on user input.
+             *
+             * @param {Object} payload - Data to update the filter.
+             * @param {string} payload.schemaName - The name of the schema (the base
+             *                                      key in the `state.filters` object
+                     *                              in which to store this data).
+             * @param {string} payload.fieldName - The name of the field that this query
+             *                                     pertains to.
+             * @param {string} payload.fieldType - The type of field represented by this
+             *                                     filter (one of 'text', 'select',
+             *                                     'min', or 'max').
+             * @param {string} payload.query - The user-inputted query value.
+             */
+            // Validate input parameters.
+            const schemaName = payload.schemaName;
+            const fieldName = payload.fieldName;
+            const fieldType = payload.fieldType;
+            const query = payload.query;
+
+            const paramDict = {
+                'schemaName': schemaName,
+                'fieldName': fieldName,
+                'fieldType': fieldType,
+                'query': query
+            };
+
+            Object.keys(paramDict).forEach(paramName => {
+                let param = paramDict[paramName];
+                if (!param) {
+                    throw new Error(`
+                        Function argument "${paramName}" is required in order
```

Comment:
I'm able to trigger this error by entering a search query in one of the text fields, clicking `Search`, and then deleting the query. After doing that, I get a console error saying `Function argument "query" is required in order to update filters.`

Reply:
Yikes, looks like two bugs colliding here. Good catch!